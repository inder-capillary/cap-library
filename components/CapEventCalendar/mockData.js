export const events = [
  {
    event_id: 1,
    title: "Event 1",
    start: "2022/4/1",
    end: "2022/6/30",
    backgroundColor: "#e9e9ea"
  },
  {
    event_id: 2,
    title: "Event 2",
    start: "2022/4/1",
    end: "2022/6/30",
    backgroundColor: "#c7e7c7"
  },
  {
    event_id: 3,
    title: "Event 3",
    start: "2022/5/4",
    end: "2022/5/29",
    backgroundColor: "#c7e7c7"
  },
  {
    event_id: 4,
    title: "Event 4",
    start: "2022/4/15",
    end: "2022/5/1",
    backgroundColor: "#feedc0"
  },
  {
    event_id: 5,
    title: "Event 5",
    start: "2022/3/15",
    end: "2022/5/16",
    backgroundColor: "#c7e7c7"
  },
  {
    event_id: 6,
    title: "Event 6",
    start: "2022/5/20",
    end: "2022/9/30",
    backgroundColor: "#feedc0"
  },
  {
    event_id: 7,
    title: "Event 7",
    start: "2022/3/16",
    end: "2022/9/29",
    backgroundColor: "#e9e9ea"
  },
  {
    event_id: 8,
    title: "Event 8",
    start: "2022/4/1",
    end: "2022/4/14",
    backgroundColor: "#c7e7c7"
  },
  {
    event_id: 9,
    title: "Event 9",
    start: "2022/4/15",
    end: "2022/4/30",
    backgroundColor: "#feedc0"
  },
  {
    event_id: 10,
    title: "Event 10",
    start: "2022/5/1",
    end: "2022/5/14",
    backgroundColor: "#e9e9ea"
  },
  {
    event_id: 11,
    title: "Event 11",
    start: "2022/5/15",
    end: "2022/5/30",
    backgroundColor: "#c7e7c7"
  },
  {
    event_id: 12,
    title: "Event 12",
    start: "2022/6/1",
    end: "2022/6/15",
    backgroundColor: "#feedc0"
  },
  {
    event_id: 13,
    title: "Event 13",
    start: "2022/6/15",
    end: "2022/6/30",
    backgroundColor: "#e9e9ea"
  },
  {
    event_id: 1212,
    title: "Event 12 copy",
    start: "2022/6/1",
    end: "2022/6/15",
    backgroundColor: "#feedc0"
  },
  {
    event_id: 1313,
    title: "Event 13 copy",
    start: "2022/6/15",
    end: "2022/6/30",
    backgroundColor: "#e9e9ea"
  },
  {
    event_id: 14,
    title: "Event 14",
    start: "2022/3/16",
    end: "2022/9/29",
    backgroundColor: "#e9e9ea"
  },
  {
    event_id: 15,
    title: "Event 15",
    start: "2022/4/1",
    end: "2022/4/14",
    backgroundColor: "#c7e7c7"
  },
  {
    event_id: 16,
    title: "Event 16",
    start: "2022/4/15",
    end: "2022/4/30",
    backgroundColor: "#feedc0"
  },
  {
    event_id: 17,
    title: "Event 17",
    start: "2022/5/1",
    end: "2022/5/14",
    backgroundColor: "#e9e9ea"
  },
  {
    event_id: 18,
    title: "Event 18",
    start: "2022/5/15",
    end: "2022/5/30",
    backgroundColor: "#c7e7c7"
  },
  {
    event_id: 19,
    title: "Event 19",
    start: "2022/6/1",
    end: "2022/6/15",
    backgroundColor: "#feedc0"
  },
  {
    event_id: 20,
    title: "Event 20",
    start: "2022/6/15",
    end: "2022/6/30",
    backgroundColor: "#e9e9ea"
  }
];
