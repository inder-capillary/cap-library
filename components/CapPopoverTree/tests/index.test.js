
import React from 'react';
import { shallow, mount } from 'enzyme';
import { CapPopoverTree } from '../index';
import { mockData } from './mockData';


jest.mock("../../CapPopover", () => ({
  __esModule: true,
  default: (props) => <div className="cap-popover" {...props}>CapPopover Mock</div>,
}));


describe('<CapPopoverTree />', () => {
  let wrapper;
  const onSelect = jest.fn();
  const { treeData, treeNode, componentProps, selectedParentKey, selectedChildKey } = mockData;
  const renderFunction = (props) => mount(<CapPopoverTree treeData={treeData} {...props} />);


  it('should toggle visibility CapPopoverTree on trigger click', () => {
    wrapper = renderFunction(componentProps);
    expect(wrapper).toMatchSnapshot();
    componentProps.visible = false;
    wrapper = renderFunction(componentProps);
    expect(wrapper).toMatchSnapshot();
  });

  const renderOverlayComponent = () => {
    const OverlayComponent = () => wrapper.find('_default').props().content;
    return shallow(<OverlayComponent />);
  };

  it('should get filtered result on search', () => {
    const event = { target: { value: 'brandCode' }};
    wrapper = renderFunction(componentProps);
    wrapper = renderOverlayComponent().find('.search-tree-node').props().onChange(event);
    expect(wrapper).toMatchSnapshot();
    event.target.value = 'on Event';
    wrapper = renderFunction(componentProps);
    wrapper = renderOverlayComponent().find('.search-tree-node').props().onChange(event);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should select the child node onSelect', () => {
    componentProps.onSelect = onSelect;
    wrapper = renderFunction(componentProps);
    wrapper = renderOverlayComponent().find('Styled(CapTree)').props().onSelect(selectedChildKey, treeNode);
    expect(onSelect).toHaveBeenCalledWith(selectedChildKey);
    expect(onSelect).toHaveBeenCalledTimes(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should expand and collapse the parent node onSelect', () => {
    treeNode.node.props.isLeafNode = false;
    treeNode.node.props.expanded = false;
    wrapper = renderFunction(componentProps);
    wrapper = renderOverlayComponent().find('Styled(CapTree)').props().onSelect(selectedParentKey, treeNode);
    expect(onSelect).toHaveBeenCalledTimes(0);
    expect(wrapper).toMatchSnapshot();
  });
});
