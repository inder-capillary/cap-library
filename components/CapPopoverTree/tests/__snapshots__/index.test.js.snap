// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CapPopoverTree /> Should expand and collapse the parent node onSelect 1`] = `undefined`;

exports[`<CapPopoverTree /> Should select the child node onSelect 1`] = `undefined`;

exports[`<CapPopoverTree /> should get filtered result on search 1`] = `undefined`;

exports[`<CapPopoverTree /> should get filtered result on search 2`] = `undefined`;

exports[`<CapPopoverTree /> should toggle visibility CapPopoverTree on trigger click 1`] = `
<CapPopoverTree
  blockNode={true}
  defaultExpandAll={false}
  placement="rightBottom"
  showIcon={true}
  switcherIcon={
    <ForwardRef(Styled(CapIcon))
      size="m"
      type="chevron-down"
    />
  }
  treeData={
    Array [
      Object {
        "children": Array [
          Object {
            "key": "skuCode",
            "title": "skuCode",
          },
          Object {
            "key": "brandCode",
            "title": "brandCode",
          },
        ],
        "icon": "survey-response",
        "key": "onEvent",
        "title": "on Event",
      },
      Object {
        "children": Array [
          Object {
            "key": "newStore",
            "title": "New store",
          },
        ],
        "icon": "bulb",
        "key": "onStore",
        "title": "on Store",
      },
    ]
  }
  trigger="click"
  visible={true}
>
  <_default
    blockNode={true}
    content={
      <CapRow
        className="search-and-tree-wrapper"
      >
        <_class
          className="search-tree-node"
          labelPosition="top"
          onChange={[Function]}
        />
        <div
          className="divider"
        />
        <ForwardRef(Styled(CapTree))
          blockNode={true}
          defaultExpandAll={false}
          expandedKeys={Array []}
          isExpanded={0}
          onExpand={[Function]}
          onSelect={[Function]}
          showIcon={true}
          switcherIcon={
            <ForwardRef(Styled(CapIcon))
              size="m"
              type="chevron-down"
            />
          }
        >
          <ContextTreeNode
            icon={
              <ForwardRef(Styled(CapIcon))
                className="tree-node-icon"
                size="s"
                type="survey-response"
              />
            }
            isLeafNode={false}
            title={
              <ForwardRef(Styled(styled.div))
                className="tree-node-title"
                type="h5"
              >
                on Event
              </ForwardRef(Styled(styled.div))>
            }
          >
            <ContextTreeNode
              isLeafNode={true}
              title={
                <ForwardRef(Styled(styled.div))
                  className="tree-node-title"
                  type="h5"
                >
                  skuCode
                </ForwardRef(Styled(styled.div))>
              }
            >
              0
            </ContextTreeNode>
            <ContextTreeNode
              isLeafNode={true}
              title={
                <ForwardRef(Styled(styled.div))
                  className="tree-node-title"
                  type="h5"
                >
                  brandCode
                </ForwardRef(Styled(styled.div))>
              }
            >
              0
            </ContextTreeNode>
          </ContextTreeNode>
          <ContextTreeNode
            icon={
              <ForwardRef(Styled(CapIcon))
                className="tree-node-icon"
                size="s"
                type="bulb"
              />
            }
            isLeafNode={false}
            title={
              <ForwardRef(Styled(styled.div))
                className="tree-node-title"
                type="h5"
              >
                on Store
              </ForwardRef(Styled(styled.div))>
            }
          >
            <ContextTreeNode
              isLeafNode={true}
              title={
                <ForwardRef(Styled(styled.div))
                  className="tree-node-title"
                  type="h5"
                >
                  New store
                </ForwardRef(Styled(styled.div))>
              }
            >
              0
            </ContextTreeNode>
          </ContextTreeNode>
        </ForwardRef(Styled(CapTree))>
      </CapRow>
    }
    overlayClassName="cap-popover-tree-v2"
    placement="rightBottom"
    showIcon={true}
    trigger="click"
    visible={true}
  >
    <div
      blockNode={true}
      className="cap-popover"
      content={
        <CapRow
          className="search-and-tree-wrapper"
        >
          <_class
            className="search-tree-node"
            labelPosition="top"
            onChange={[Function]}
          />
          <div
            className="divider"
          />
          <ForwardRef(Styled(CapTree))
            blockNode={true}
            defaultExpandAll={false}
            expandedKeys={Array []}
            isExpanded={0}
            onExpand={[Function]}
            onSelect={[Function]}
            showIcon={true}
            switcherIcon={
              <ForwardRef(Styled(CapIcon))
                size="m"
                type="chevron-down"
              />
            }
          >
            <ContextTreeNode
              icon={
                <ForwardRef(Styled(CapIcon))
                  className="tree-node-icon"
                  size="s"
                  type="survey-response"
                />
              }
              isLeafNode={false}
              title={
                <ForwardRef(Styled(styled.div))
                  className="tree-node-title"
                  type="h5"
                >
                  on Event
                </ForwardRef(Styled(styled.div))>
              }
            >
              <ContextTreeNode
                isLeafNode={true}
                title={
                  <ForwardRef(Styled(styled.div))
                    className="tree-node-title"
                    type="h5"
                  >
                    skuCode
                  </ForwardRef(Styled(styled.div))>
                }
              >
                0
              </ContextTreeNode>
              <ContextTreeNode
                isLeafNode={true}
                title={
                  <ForwardRef(Styled(styled.div))
                    className="tree-node-title"
                    type="h5"
                  >
                    brandCode
                  </ForwardRef(Styled(styled.div))>
                }
              >
                0
              </ContextTreeNode>
            </ContextTreeNode>
            <ContextTreeNode
              icon={
                <ForwardRef(Styled(CapIcon))
                  className="tree-node-icon"
                  size="s"
                  type="bulb"
                />
              }
              isLeafNode={false}
              title={
                <ForwardRef(Styled(styled.div))
                  className="tree-node-title"
                  type="h5"
                >
                  on Store
                </ForwardRef(Styled(styled.div))>
              }
            >
              <ContextTreeNode
                isLeafNode={true}
                title={
                  <ForwardRef(Styled(styled.div))
                    className="tree-node-title"
                    type="h5"
                  >
                    New store
                  </ForwardRef(Styled(styled.div))>
                }
              >
                0
              </ContextTreeNode>
            </ContextTreeNode>
          </ForwardRef(Styled(CapTree))>
        </CapRow>
      }
      overlayClassName="cap-popover-tree-v2"
      placement="rightBottom"
      showIcon={true}
      trigger="click"
      visible={true}
    >
      CapPopover Mock
    </div>
  </_default>
</CapPopoverTree>
`;

exports[`<CapPopoverTree /> should toggle visibility CapPopoverTree on trigger click 2`] = `
<CapPopoverTree
  blockNode={true}
  defaultExpandAll={false}
  placement="rightBottom"
  showIcon={true}
  switcherIcon={
    <ForwardRef(Styled(CapIcon))
      size="m"
      type="chevron-down"
    />
  }
  treeData={
    Array [
      Object {
        "children": Array [
          Object {
            "key": "skuCode",
            "title": "skuCode",
          },
          Object {
            "key": "brandCode",
            "title": "brandCode",
          },
        ],
        "icon": "survey-response",
        "key": "onEvent",
        "title": "on Event",
      },
      Object {
        "children": Array [
          Object {
            "key": "newStore",
            "title": "New store",
          },
        ],
        "icon": "bulb",
        "key": "onStore",
        "title": "on Store",
      },
    ]
  }
  trigger="click"
  visible={false}
>
  <_default
    blockNode={true}
    content={
      <CapRow
        className="search-and-tree-wrapper"
      >
        <_class
          className="search-tree-node"
          labelPosition="top"
          onChange={[Function]}
        />
        <div
          className="divider"
        />
        <ForwardRef(Styled(CapTree))
          blockNode={true}
          defaultExpandAll={false}
          expandedKeys={Array []}
          isExpanded={0}
          onExpand={[Function]}
          onSelect={[Function]}
          showIcon={true}
          switcherIcon={
            <ForwardRef(Styled(CapIcon))
              size="m"
              type="chevron-down"
            />
          }
        >
          <ContextTreeNode
            icon={
              <ForwardRef(Styled(CapIcon))
                className="tree-node-icon"
                size="s"
                type="survey-response"
              />
            }
            isLeafNode={false}
            title={
              <ForwardRef(Styled(styled.div))
                className="tree-node-title"
                type="h5"
              >
                on Event
              </ForwardRef(Styled(styled.div))>
            }
          >
            <ContextTreeNode
              isLeafNode={true}
              title={
                <ForwardRef(Styled(styled.div))
                  className="tree-node-title"
                  type="h5"
                >
                  skuCode
                </ForwardRef(Styled(styled.div))>
              }
            >
              0
            </ContextTreeNode>
            <ContextTreeNode
              isLeafNode={true}
              title={
                <ForwardRef(Styled(styled.div))
                  className="tree-node-title"
                  type="h5"
                >
                  brandCode
                </ForwardRef(Styled(styled.div))>
              }
            >
              0
            </ContextTreeNode>
          </ContextTreeNode>
          <ContextTreeNode
            icon={
              <ForwardRef(Styled(CapIcon))
                className="tree-node-icon"
                size="s"
                type="bulb"
              />
            }
            isLeafNode={false}
            title={
              <ForwardRef(Styled(styled.div))
                className="tree-node-title"
                type="h5"
              >
                on Store
              </ForwardRef(Styled(styled.div))>
            }
          >
            <ContextTreeNode
              isLeafNode={true}
              title={
                <ForwardRef(Styled(styled.div))
                  className="tree-node-title"
                  type="h5"
                >
                  New store
                </ForwardRef(Styled(styled.div))>
              }
            >
              0
            </ContextTreeNode>
          </ContextTreeNode>
        </ForwardRef(Styled(CapTree))>
      </CapRow>
    }
    overlayClassName="cap-popover-tree-v2"
    placement="rightBottom"
    showIcon={true}
    trigger="click"
    visible={false}
  >
    <div
      blockNode={true}
      className="cap-popover"
      content={
        <CapRow
          className="search-and-tree-wrapper"
        >
          <_class
            className="search-tree-node"
            labelPosition="top"
            onChange={[Function]}
          />
          <div
            className="divider"
          />
          <ForwardRef(Styled(CapTree))
            blockNode={true}
            defaultExpandAll={false}
            expandedKeys={Array []}
            isExpanded={0}
            onExpand={[Function]}
            onSelect={[Function]}
            showIcon={true}
            switcherIcon={
              <ForwardRef(Styled(CapIcon))
                size="m"
                type="chevron-down"
              />
            }
          >
            <ContextTreeNode
              icon={
                <ForwardRef(Styled(CapIcon))
                  className="tree-node-icon"
                  size="s"
                  type="survey-response"
                />
              }
              isLeafNode={false}
              title={
                <ForwardRef(Styled(styled.div))
                  className="tree-node-title"
                  type="h5"
                >
                  on Event
                </ForwardRef(Styled(styled.div))>
              }
            >
              <ContextTreeNode
                isLeafNode={true}
                title={
                  <ForwardRef(Styled(styled.div))
                    className="tree-node-title"
                    type="h5"
                  >
                    skuCode
                  </ForwardRef(Styled(styled.div))>
                }
              >
                0
              </ContextTreeNode>
              <ContextTreeNode
                isLeafNode={true}
                title={
                  <ForwardRef(Styled(styled.div))
                    className="tree-node-title"
                    type="h5"
                  >
                    brandCode
                  </ForwardRef(Styled(styled.div))>
                }
              >
                0
              </ContextTreeNode>
            </ContextTreeNode>
            <ContextTreeNode
              icon={
                <ForwardRef(Styled(CapIcon))
                  className="tree-node-icon"
                  size="s"
                  type="bulb"
                />
              }
              isLeafNode={false}
              title={
                <ForwardRef(Styled(styled.div))
                  className="tree-node-title"
                  type="h5"
                >
                  on Store
                </ForwardRef(Styled(styled.div))>
              }
            >
              <ContextTreeNode
                isLeafNode={true}
                title={
                  <ForwardRef(Styled(styled.div))
                    className="tree-node-title"
                    type="h5"
                  >
                    New store
                  </ForwardRef(Styled(styled.div))>
                }
              >
                0
              </ContextTreeNode>
            </ContextTreeNode>
          </ForwardRef(Styled(CapTree))>
        </CapRow>
      }
      overlayClassName="cap-popover-tree-v2"
      placement="rightBottom"
      showIcon={true}
      trigger="click"
      visible={false}
    >
      CapPopover Mock
    </div>
  </_default>
</CapPopoverTree>
`;
